drivers/adc_ng/adc_ng.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_adc_ng
+ * @{
+ *
+ * @file
+ * @brief       Implementation of common functions of the ADC API
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ * @}
+ */
+
+#include <errno.h>
+
+#include "adc_ng.h"
+
+#if (ADC_NG_NUMOF == 1) && defined(MODULE_PERIPH_ADC_NG)
```

Comment:
```suggestion
#if (ADC_NG_NUMOF == 1) && MODULE_PERIPH_ADC_NG
```
could be used instead since an undefined macro yields 0 in expressions by definition.

Reply:
Sure? It wasn't like this a month ago. There was a heated discussion for adding `IS_USED(MODULE_FOO)` to get `0` in case either `MODULE_FOO` is `0` or not defined.

Reply:
> Sure? It wasn't like this a month ago. There was a heated discussion for adding `IS_USED(MODULE_FOO)` to get `0` in case either `MODULE_FOO` is `0` or not defined.

yeah, I figured that out in the mentioned discussion. An undefined macro resolves to "0" only in preprocessor expressions. With ```IS_USED()``` it can be used in C expressions.

Reply:
> An undefined macro resolves to "0" only in preprocessor expressions.

Exactly. So if there are no design rules against it (???), `#if MODULE_FOO` could be used especially in long expressions to keep them shorter and avoid cascades of `#if defined() && !defined() && defined()`.